File: Data Structures & Algorithms/largest-substring-between-two-equal-characters/submission-1.py
class Solution:
    def maxLengthBetweenEqualCharacters(self, s: str) -> int:
        char = {}
        res = -1

        for i, c in enumerate(s):
            if c in char:
                res = max(res, i - char[c] - 1)
            else:
                char[c] = i
        return res
        

File: Data Structures & Algorithms/sliding-window-maximum/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    maxSlidingWindow(nums, k) {
        // move with k and take max
        let res = []
        // slice the block
        for (let i = 0; i + k <= nums.length; i++) {
            let block = nums.slice(i, i + k)
            res.push(Math.max(...block))
        }

        console.log(res, "_res")
        return res
    }
}
